src/pages/events.tsx

```diff
@@ -58,22 +59,72 @@ const filterEvent = (
   return true;
 };
 
+const CommunityOptions = [
+  { value: 'all', label: 'All communities' },
+  { value: 'general', label: 'General' },
+  { value: 'ai', label: 'AI' },
+  { value: 'cyber', label: 'Cyber' },
+  { value: 'design', label: 'Design' },
+  { value: 'hack', label: 'Hack' },
+];
+
+const TimeOptions = [
+  { value: 'upcoming', label: 'Upcoming' },
+  { value: 'past-week', label: 'Past week' },
+  { value: 'past-month', label: 'Past month' },
+  { value: 'past-year', label: 'Past year' },
+  { value: 'all-time', label: 'All time' },
+];
+
+const AttendanceOptions = [
+  { value: 'all', label: 'Any attendance' },
+  { value: 'attended', label: 'Attended' },
+  { value: 'not-attended', label: 'Not attended' },
+];
+
 const ROWS_PER_PAGE = 25;
 const EventsPage = ({ events, attendances }: EventsPageProps) => {
   const [page, setPage] = useState(0);
   const [communityFilter, setCommunityFilter] = useState('all');
-  const [dateFilter, setDateFilter] = useState('upcoming');
+  const [timeFilter, setTimeFilter] = useState('all-time');
   const [attendedFilter, setAttendedFilter] = useState('all');
   const [query, setQuery] = useState('');
 
   const years = useMemo(getYears, []);
 
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const community = searchParams.get('community');
+    if (CommunityOptions.some(({ value }) => value === community)) {
+      setCommunityFilter(community as string);
+    }
+
+    const time = searchParams.get('time');
+    if (
+      TimeOptions.some(({ value }) => value === time) ||
+      years.some(({ value }) => value === time)
+    ) {
+      setTimeFilter(time as string);
+    }
+
+    const attendance = searchParams.get('attendance');
+    if (AttendanceOptions.some(({ value }) => value === attendance)) {
+      setAttendedFilter(attendance as string);
+    }
+
+    const query = searchParams.get('query');
+    if (query !== null) {
+      setQuery(query);
+    }
+  }, [searchParams, years]);
```

Comment:
I'm getting a layout shift from this because it reads the URL parameters in a `useEffect`

Since `searchParams` and `years` both won't change after loading the page, I think you could try something like

```ts
const { defaultCommunityFilter, defaultTimeFilter, defaultAttendedFilter, defaultQuery } = useMemo(() => getDefaultFilters(searchParams, years), [searchParams, years]);
const [communityFilter, setCommunityFilter] = useState(defaultCommunityFilter);
const [timeFilter, setTimeFilter] = useState(defaultTimeFilter);
const [attendedFilter, setAttendedFilter] = useState(defaultAttendedFilter);
const [query, setQuery] = useState(defaultQuery);
```